### ISSUE TITLE
Date.strptime returns wrong month for %Y

### ISSUE THREAD

**Issue by krzyzak** [travisjeffery/timecop#377]
Hey,

Today I found issue with `Date.strptime`: it falls back to current month, rather than returning Jan 1st, when parsing just `%Y`:

```ruby
Date.strptime('2000', "%Y")
=> Sat, 01 Jan 2000
require 'timecop'
=> true
Date.strptime('2000', "%Y")
=> Tue, 01 Feb 2000
```

Seems that the issue lies in <a href="https://github.com/travisjeffery/timecop/blob/master/lib/timecop/time_extensions.rb#L54-L56">time_extensions.rb</a>.

**Comment by augustoccesar** [comment]
I was looking into opening a PR for this issue, but from the short investigation that I did it seems that this is a very explicit expected behaviour.

https://github.com/travisjeffery/timecop/blob/8ca1018694fd9f906c95d92e4b90e5fe1be4429e/lib/timecop/time_extensions.rb#L53-L56

I guess to change to the expected behaviour described in this issue it would need to change to something like
```ruby
d = Date._strptime(str, fmt) || Date.strptime_without_mock_date(str, fmt)
now = Time.now.to_date
year = d[:year] || now.year
mon = d[:mon] || 1
```
which feels like a large change from the perspective of expectation of how this method works (on timecop).

So I think that the main question is: Should the timecop `Date.strptime` behaves like the Ruby `Date.strptime`, or this an expected/desired behaviour to default to the current month?

**Comment by joshuacronemeyer** [comment]
Thanks guys. The Timecop implementation of strptime is a work in progress. Our implementation is incomplete and we're slowly chipping away at it. It should behave like the ruby implementation.

checkout https://github.com/travisjeffery/timecop/blob/8ca1018694fd9f906c95d92e4b90e5fe1be4429e/test/date_strptime_scenarios.rb#L68-L81 to see a bunch of time format stuff we haven't handled yet. The scenario highlighted by this issue is just one more to add to that list.

**Comment by joshuacronemeyer** [comment]
@krzyzak and @augustoccesar i'd appreciate if you'd have a look at my PR https://github.com/travisjeffery/timecop/pull/379 for this issue. I'm open to suggestions on ways to better implement this.

**Comment by krzyzak** [comment]
@joshuacronemeyer yes, that indeed fixes my issue. Is there an option to release new gem version?

**Comment by joshuacronemeyer** [comment]
@krzyzak thanks for asking. I just released 0.9.5 that includes this change. Would appreciate you bumping your version and testing.
